src/common/map/MapDirective.js

```diff
@@ -53,15 +53,44 @@
 
             scope.$watch('layers', function(layers) {
               if (layers && map) {
-                var layerLength = map.getLayers().getLength();
-                for (var i = 1; i < layerLength; i++) {
-                  map.removeLayer(map.getLayers().getArray()[1]);
+                deleteNoCartLayers();
+                addAlllayers(layers);
+              }
+            });
+
+            function deleteNoCartLayers() {
```

Comment:
there is nothing with the cart involved in this method. why does it have cart in the name?